### PR TITLE
Add new client

### PR DESCRIPTION
**Pull request by joukevandermaas**
This PR moves the spot where the current token is kept to the NfieldConnection, while keeping the logic for the refresh in the client.

It also introduces a new client, to be used when doing Active Directory authentication.

### CODE REVIEW

File: Library/Infrastructure/ActiveDirectoryNfieldHttpClient.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+using Newtonsoft.Json;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Nfield.Infrastructure
+{
+    internal sealed class ActiveDirectoryNfieldHttpClient : INfieldHttpClient
```
Comment (LNieuwmegen) [inline Library/Infrastructure/ActiveDirectoryNfieldHttpClient.cs]: Maybe we should call this `BearerTokenHttpClient`?

File: Library/Infrastructure/DefaultNfieldHttpClient.cs [hunk 2]
```diff
@@ -29,12 +29,16 @@ namespace Nfield.Infrastructure
     /// A wrapper around <see cref="HttpClient"/> that also adds an authentication header to the response
     /// if it was received in the request.
     /// </summary>
-    internal sealed class NfieldHttpClient : INfieldHttpClient
+    internal sealed class DefaultNfieldHttpClient : INfieldHttpClient
```
Comment (LNieuwmegen) [inline Library/Infrastructure/DefaultNfieldHttpClient.cs]: `NfieldTokenHttpClient`?

File: Library/Infrastructure/ActiveDirectoryNfieldHttpClient.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            _domainName = domainName;
+            _token = token;
+        }
+
+        public override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request)
+        {
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", _token);
+            request.Headers.Add("X-Nfield-Domain", _domainName);
+
```
Comment (mvdr) [inline Library/Infrastructure/ActiveDirectoryNfieldHttpClient.cs]: Not for every call a domain header is needed (domainassignments endpoint), so probably check if _domainName has a value and based on that add a header or leave it out.

Comment (joukevandermaas) [reply Library/Infrastructure/ActiveDirectoryNfieldHttpClient.cs]: we don't (currently) have the ability to make domain assignments through the SDK, so for all currently supported endpoints it *is* needed.

if in the future this changes, let's address that problem then.

**Pull request merged**